lib/smart_answer/calculators/arrested_abroad.rb

```diff
@@ -26,16 +24,80 @@ def generate_url_for_download(country, field, text)
       end
     end
 
-    def self.prisoner_packs
-      @prisoner_packs ||= YAML.load_file(Rails.root.join("config/smart_answers/prisoner_packs.yml"))
-    end
-
     def countries_with_regions
       %w[cyprus]
     end
 
     def get_country_regions(slug)
-      @data.find { |c| c["slug"] == slug }["regions"]
+      PRISONER_PACKS.find { |c| c["slug"] == slug }["regions"]
+    end
+
+    def location
+      @location ||= WorldLocation.find(country)
+      raise InvalidResponse unless @location
+
+      @location
+    end
+
+    def organisation
+      location.fco_organisation
+    end
+
+    def country_name
+      location.name
+    end
+
+    def pdf
+      generate_url_for_download(country, "pdf", "Prisoner pack for #{country_name}")
```

Comment:
If `generate_url_for_download` is rewritten as I've suggested, `country` no longer needs to be passed into the method.